src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -122,15 +122,31 @@ private static int ComputeSymbolMatchPriority(ISymbol symbol)
             var position = SymbolCompletionItem.GetContextPosition(item);
             var name = SymbolCompletionItem.GetSymbolName(item);
             var kind = SymbolCompletionItem.GetKind(item);
+            bool.TryParse(SymbolCompletionItem.GetSymbolIsGeneric(item), out var isGeneric);
             var relatedDocumentIds = document.Project.Solution.GetRelatedDocumentIds(document.Id).Concat(document.Id);
             var options = document.Project.Solution.Workspace.Options;
             var totalSymbols = await base.GetPerContextSymbols(document, position, options, relatedDocumentIds, preselect: false, cancellationToken: cancellationToken).ConfigureAwait(false);
             foreach (var (documentId, syntaxContext, symbols) in totalSymbols)
             {
-                var bestSymbols = symbols.Where(s => kind != null && s.Kind == kind && s.Name == name).ToImmutableArray();
+                var bestSymbols = symbols.Where(s => kind != null && s.Kind == kind && s.Name == name && isGeneric == (s.GetArity() != 0));
```

Comment:
oops, sorry, didn't mean to commit to your branch directly.  that was supposed to be a comment.

Reply:
I was not aware there is a `GetArity()` extension method, fixed. Thanks for the suggestions!